### PR TITLE
Move ValidateConfig from configcheck to configtest

### PR DESCRIPTION
**Pull request by alexperez52**
**Description:** 
This PR moves ValidateConfig from being used in configcheck to configtest and updates the import path usages.

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector/issues/3875

### CODE REVIEW

File: config/configtest/configtest.go [hunk 1]
```diff
@@ -40,3 +49,131 @@ func LoadConfigAndValidate(fileName string, factories component.Factories) (*con
 	}
 	return cfg, cfg.Validate()
 }
+
+// ValidateConfig enforces that given configuration object is following the patterns
+// used by the collector. This ensures consistency between different implementations
+// of components and extensions. It is recommended for implementers of components
+// to call this function on their tests passing the default configuration of the
+// component factory.
+func ValidateConfig(config interface{}) error {
```
Comment (bogdandrutu) [inline config/configtest/configtest.go:58]: s/ValidateConfig/CheckConfigStruct?

Comment (alexperez52) [reply config/configtest/configtest.go]: Updated

**Review by bogdandrutu — APPROVED** [review]

File: processor/batchprocessor/config_test.go [hunk 2]
```diff
@@ -56,7 +56,7 @@ func TestLoadConfig(t *testing.T) {
 		})
 }
 
-func TestValidateConfig_DefaultBatchMaxSize(t *testing.T) {
+func TestCheckConfigStruct_DefaultBatchMaxSize(t *testing.T) {
```
Comment (bogdandrutu) [inline processor/batchprocessor/config_test.go:59]: Why this change?

File: processor/batchprocessor/config_test.go [hunk 3]
```diff
@@ -65,7 +65,7 @@ func TestValidateConfig_DefaultBatchMaxSize(t *testing.T) {
 	assert.NoError(t, cfg.Validate())
 }
 
-func TestValidateConfig_ValidBatchSizes(t *testing.T) {
+func TestCheckConfigStruct_ValidBatchSizes(t *testing.T) {
```
Comment (bogdandrutu) [inline processor/batchprocessor/config_test.go:68]: Why this change?

File: processor/batchprocessor/config_test.go [hunk 4]
```diff
@@ -75,7 +75,7 @@ func TestValidateConfig_ValidBatchSizes(t *testing.T) {
 
 }
 
-func TestValidateConfig_InvalidBatchSize(t *testing.T) {
+func TestCheckConfigStruct_InvalidBatchSize(t *testing.T) {
```
Comment (bogdandrutu) [inline processor/batchprocessor/config_test.go:78]: Why this change?

File: config/configcheck/configcheck.go [hunk 5]
```diff
@@ -15,171 +15,36 @@
...
 )
 
-// The regular expression for valid config field tag.
-var configFieldTagRegExp = regexp.MustCompile("^[a-z0-9][a-z0-9_]*$")
-
-// ValidateConfigFromFactories checks if all configurations for the given factories
+// CheckConfigStructFromFactories checks if all configurations for the given factories
 // are satisfying the patterns used by the collector.
-func ValidateConfigFromFactories(factories component.Factories) error {
+func CheckConfigStructFromFactories(factories component.Factories) error {
```
Comment (bogdandrutu) [inline config/configcheck/configcheck.go:25]: Why this name change?

File: config/configtest/configtest.go (hunk 1 above)
Comment (bogdandrutu) [reply config/configtest/configtest.go:58]: @Aneurysm9 I asked for the change only of this function name not every time "ValidateConfig" is used.

**Comment by alolita** [comment]
@Aneurysm9 @codeboten can you please review this PR?

**Comment by bogdandrutu** [comment]
@alolita I cannot since I asked for changes and they are not "resolved" - either implemented or explained why not to.

**Comment by hyunuk** [comment]
@bogdandrutu changed only for ValidateConfig() to CheckConfigStruct()

**Review by bogdandrutu — APPROVED** [review]

**Comment by alolita** [comment]
@Aneurysm9 can you please review too? So that @bogdandrutu can merge :-)

File: config/configtest/configtest_test.go [hunk 6]
```diff
@@ -79,3 +81,138 @@ func TestLoadConfigAndValidate(t *testing.T) {
 
 	assert.Equal(t, cfg, cfgValidate)
 }
+
+func TestValidateConfigPointerAndValue(t *testing.T) {
```
Comment (Aneurysm9) [inline config/configtest/configtest_test.go:85]:
```suggestion
func TestCheckConfigStructPointerAndValue(t *testing.T) {
```

File: config/configtest/configtest_test.go [hunk 7]
```diff
@@ -79,3 +81,138 @@ func TestLoadConfigAndValidate(t *testing.T) {
...
+	config := struct {
+		SomeFiled string `mapstructure:"test"`
+	}{}
+	assert.NoError(t, CheckConfigStruct(config))
+	assert.NoError(t, CheckConfigStruct(&config))
+}
+
+func TestValidateConfig(t *testing.T) {
```
Comment (Aneurysm9) [inline config/configtest/configtest_test.go:93]:
```suggestion
func TestCheckConfigStruct(t *testing.T) {
```

**Review by Aneurysm9 — APPROVED** [review]
Other than the two test names this LGTM.

**Review by alolita — APPROVED** [review]

**Comment by alolita** [comment]
@bogdandrutu please merge
